colin-api/src/colin_api/models/corp_party.py

```diff
@@ -104,8 +106,27 @@ def _build_parties_list(cls, cursor, event_id: int = None):
                     party.cessation_date = None
 
                 party_list.append(party)
+        return cls.group_parties(party_list)
 
-        return party_list
+    @classmethod
+    def group_parties(cls, parties):
+        """Group parties based on roles."""
+        grouped_list = []
+        role_func = (lambda x: x.officer['firstName'] + ' ' + x.officer['middleInitial'] + ' ' + x.officer['lastName']
+                     + ' ' + x.officer['orgName'])  # noqa: E731;
+
+        parties_dict = {k: list(v) for k, v in itertools.groupby(parties, key=role_func)}
+        role_dict = {v: k for k, v in cls.role_types.items()}
+
+        for k, v in parties_dict.items():  # pylint: disable=unused-variable;
+            party = v[0]
+            roles = []
+            for i in v:
+                role = i.role_type
+                roles.append({'roleType': role_dict[role]})
+            party.roles = roles
+            grouped_list.append(party)
+        return grouped_list
```

Comment:
It's pretty hard to read this because of the variable names used here (x , k, v don't give me any idea of what you're looping through etc.). Would help a lot if you added in a couple short comments to describe what's going on

Reply:
should be better now  (the k,v are short for key, value, and the x just represents the captured element in the lambda)...most other languages I've worked with seem to encourage this type of convention for inline/lambda functions

Reply:
Oh I see that makes sense. Thanks for adding the documentation